### ISSUE TITLE
Media: Attachment pages not displaying image comments

### ISSUE THREAD

**Issue by davipontesblog** [Automattic/wp-calypso#56790]
### Quick summary

I am not quite sure how reproducible this is but logging it anyway since I just helped a user with this. Please be aware of the very specific setup below for this user's site.

On this site, the comments were not showing, unless under `Media > Click image > Edit more details` I clicked `Screen Options` and "showed" both `Comments` and `Discussion`. Somehow, this triggered the attachment page to show the comments. I am not sure why, because as far as I know the "Screen Options" are only supposed to affect the admin details page for the images on the site.

### Steps to reproduce

- On a simple site, activate the "Twenty Ten" theme.
- Publish a page with a Classic block, and a Gallery in it.
- Under WP Admin > Settings > Media, make sure `Enable carousel` is toggled OFF.
- Add a comment to one of the images, whether on the attachment page live, or in Media > Click image > Edit more details.

### What you expected to happen

- Open live attachment page, and see that the comment is showing.

### What actually happened

On this site, the comments were not showing, unless under `Media > Click image > Edit more details` I clicked `Screen Options` and "showed" both `Comments` and `Discussion`. Somehow, this triggered the attachment page to show the comments. I am not sure why, because as far as I know the "Screen Options" are only supposed to affect the admin details page for the images on the site.

### Context

User report.

### Reproducibility

Once

### Severity

One

### Available workarounds?

Yes, easy to implement

### Workaround details

Under `Media > Click image > Edit more details` click `Screen Options` and "show" both `Comments` and `Discussion`. Somehow, this triggers the attachment page to show the comments. I am not sure why, because as far as I know the "Screen Options" are only supposed to affect the admin details page for the images on the site.

**Comment by davipontesblog** [comment]
User report: 4349501-zd-woothemes

**Comment by kosiew** [comment]
Hi @davipontesblog ,

Thanks for reporting this and the detailed steps.

At `Media > Click image > Edit more details I clicked Screen Options and "showed" both Comments and Discussion`, it opens to the edit screen similar to editing posts. 

![Markup on 2021-10-12 at 17:15:02](https://user-images.githubusercontent.com/29057562/136928058-b695ac61-24b6-4d87-ba4b-0207f7bc9235.png)

I also see the same on my test site:
![Markup on 2021-10-12 at 17:19:15](https://user-images.githubusercontent.com/29057562/136928685-bd06fcdc-8277-4562-bd8f-adf43a785041.png)

So if those options are disabled, the comments will be disabled as well.
Great job in digging and finding this!


Closing this as it is not a bug.
